src/Classes/Utils/ColorUtils.ts

```diff
@@ -0,0 +1,199 @@
+import { IColor } from "../../Interfaces/Options/Particles/IColor";
+import { IRgb } from "../../Interfaces/IRgb";
+import { IRgba } from "../../Interfaces/IRgba";
+import { IHsl } from "../../Interfaces/IHsl";
+import { IHsla } from "../../Interfaces/IHsla";
+import { Utils } from "./Utils";
+
+export class ColorUtils {
+    /**
+     * Gets the particles color
+     * @param color the input color to convert in [[IRgb]] object
+     */
+    public static colorToRgb(color: IColor): IRgb | undefined {
+        let res: IRgb | undefined;
+
+        if (typeof (color.value) === "string") {
+            if (color.value === "random") {
+                res = {
+                    b: Math.floor(Math.random() * 256),
+                    g: Math.floor(Math.random() * 256),
+                    r: Math.floor(Math.random() * 256),
+                };
+            } else {
+                res = ColorUtils.stringToRgb(color.value);
+            }
+        } else {
+            if (color.value instanceof Array) {
+                const arr = color.value as string[];
+                const colorSelected = color.value[Math.floor(Math.random() * arr.length)];
+
+                res = ColorUtils.stringToRgb(colorSelected);
+            } else {
+                const rgbColor = color.value as IRgb;
+
+                if (rgbColor.r !== undefined) {
+                    res = rgbColor;
+                }
+
+                const hslColor = color.value as IHsl;
+
+                if (hslColor.h !== undefined) {
+                    res = ColorUtils.hslToRgb(hslColor);
+                }
+            }
+        }
+
+        return res;
+    }
+
+    public static stringToAlpha(input: string) {
+        return ColorUtils.stringToRgba(input)?.a;
+    }
+
+    /**
+     * Converts hexadecimal string (HTML color code) in a [[IRgb]] object
+     * @param input the hexadecimal string (#f70 or #ff7700)
+     */
+    public static stringToRgb(input: string): IRgb | undefined {
+        return ColorUtils.stringToRgba(input);
+    }
+
+    private static stringToRgba(input: string): IRgba | undefined {
+        if (input.startsWith('rgb')) {
+            const regex = /rgba?\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*(,\s*([\d\.]+)\s*)?\)/i;
+            const result = regex.exec(input);
+
+            return result ? {
+                a: parseInt(result[4]),
+                b: parseInt(result[3]),
+                g: parseInt(result[2]),
+                r: parseInt(result[1]),
+            } : undefined;
+        } else if (input.startsWith('hsl')) {
+            const regex = /hsla?\(\s*(\d+)\s*,\s*(\d+)\%\s*,\s*(\d+)\%\s*(,\s*([\d\.]+)\s*)?\)/i;
+            const result = regex.exec(input);
+
+            return result ? ColorUtils.hslaToRgba({
+                a: parseInt(result[4]),
+                l: parseInt(result[3]),
+                s: parseInt(result[2]),
+                h: parseInt(result[1]),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The key 'h' is not sorted alphabetically](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5216054)